File: 69-sqrtx/69-sqrtx.java
class Solution {
    public int mySqrt(int x) {
        double guess = x/2d;
        double lastVal = (guess + x/guess)/2d; 
        while (true) {
            double currentVal = (lastVal + x/lastVal)/2d;
            if ((int)currentVal == (int)lastVal) {
                break;
            } 
            lastVal = currentVal;
        }
        
        return (int)lastVal;
        
    }
}

File: 69-sqrtx/69-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        int mid,low,high,temp,ans;
        low = 1;
        high = x/2;
        if(x==0||x==1)
            return x;
        while(low<=high)
        {
            mid = (low+high)/2;
            temp=trunc(x/mid);
            if(temp==mid)
               return mid;
            else if(temp>mid) {
                low=mid+1;
                ans=mid;
            }
            else
                high=mid-1;
        }
        return ans;
    }
};